File: Patterns/Composite Pattern/Cpp/Menu/MenuComponent.cpp
#pragma once

#include <string>
#include <stdexcept>

using namespace std;

class MenuComponent {
public:
    virtual void add(MenuComponent* menuComponent) {
        throw runtime_error("Unsupported operation exception.\n");
    }

    virtual void remove(MenuComponent* menuComponent) {
        throw runtime_error("Unsupported operation exception.\n");
    }

    virtual MenuComponent* getChild(int i) {
        throw runtime_error("Unsupported operation exception.\n");
    }

    virtual string getName() {
        throw runtime_error("Unsupported operation exception.\n");
    }

    virtual string getDescription() {
        throw runtime_error("Unsupported operation exception.\n");
    }

    virtual bool isVegetarian() {
        throw runtime_error("Unsupported operation exception.\n");
    }

    virtual double getPrice() {
        throw runtime_error("Unsupported operation exception.\n");
    }

    virtual void print() {
        throw runtime_error("Unsupported operation exception.\n");
    }
};

File: Patterns/Composite Pattern/CSharp/Menu/MenuComponent.cs
using System;
using System.Collections.Generic;

namespace Menu
{
    public class MenuComponent
    {
        public virtual void Add(MenuComponent menuComponent) {}

        public virtual void Remove(MenuComponent menuComponent) {}

        public virtual MenuComponent? GetChild(int i) { return null; }

        public virtual string? Name { get; set; }

        public virtual string? Description { get; set; }

        public virtual bool Vegetarian { get; set; }

        public virtual double Price { get; set; }

        public virtual void Print() {}
    }
}